swaggen/src/main/java/utils/io/FileMapper.java

```diff
@@ -38,14 +42,59 @@
 	 * 
 	 * @param filename the name of the YAML file
 	 * @param object the Java object
-	 * @throws JsonGenerationException
-	 * @throws JsonMappingException
-	 * @throws IOException
+	 * @param <T> The type
+	 * 
+	 * @throws JsonGenerationException JsonGenerationException
+	 * @throws JsonMappingException JsonMappingException
+	 * @throws IOException IOException
 	 */
 	public static <T> void classToYaml(String filename, T object)
 			throws JsonGenerationException, JsonMappingException, IOException {
 		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
 		objectMapper.writeValue(new File(filename), object);
 	}
 
+	/**
+	 * Maps a Swagger object to a YAML file.
+	 * 
+	 * @param path The file path of the swagger file
+	 * @param newSwagger the Swagger
+	 * 
+	 * @throws JsonGenerationException JsonGenerationException
+	 * @throws JsonMappingException JsonMappingException
+	 * @throws IOException IOException
+	 */
+	public static void classToYaml(String path, Swagger newSwagger)
+		throws JsonGenerationException, JsonMappingException, IOException {
+		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+		File file = new File(path.substring(1));
+
+		checkExistingEndpoints(newSwagger, file, objectMapper);
+		objectMapper.writeValue(file, newSwagger);
+	}
+
+	/**
+	 * Checks for existing endpoints in yaml files and if matched, add the existing
+	 * request methods to the new Swagger file.
+	 * @param swagger the swagger object containing all the endpoints
+	 * @param file file object, containing where it is supposed to be placed
+	 * @param objectMapper the ObjectMapper
+	 * @throws JsonGenerationException
+	 * @throws JsonMappingException
+	 * @throws IOException
+	 */
+	private static void checkExistingEndpoints(Swagger swagger, File file, ObjectMapper objectMapper) 

```

Comment:
Quick question (not necessarily an issue): why are we reading YAML files? We should take in data from the annotation and write that to YAML. What is the purpose of reading YAML files?

Reply:
We are reading YAML files because each classes's endpoints are written individually so if there are two classes with the same path, one would overwrite the other. Reading the YAML files if the file already exists allows one class to append the other class in terms of their request methods.

Reply:
Yeah, that is what I thought might be going on. This seems really inefficient: reading and writing to files multiple times. More importantly, it seems more complex than necessary. Is there a reason that we are not collecting all of the endpoints from all of the classes before writing the YAML?

Reply:
No there was no reason, it is now fixed